src/qumin/representations/segments.py

```diff
@@ -37,27 +37,33 @@ class Form(str):
 
     Sounds might be more than one character long.
     Forms are strings, they are segmented at the object creation.
-    They have a tokens attribute, which is a tuple of phonemes.
+
+    Attributes:
+        tokens (Tuple): Tuple of phonemes contained in this form
+        id (str): form_id of the corresponding form according to the Paralex package.
+            If unknown, `None` will be assigned.
     """
 
-    def __new__(cls, string):
+    def __new__(cls, string, form_id=None):
         tokens = Inventory._segmenter.findall(string)
         tokens = tuple(Inventory._normalization.get(c, c) for c in tokens)
         if Inventory._legal_str.fullmatch("".join(tokens)) is None:
             raise ValueError("Unknown sound in: " + repr(string))
         self = str.__new__(cls, " ".join(tokens) + " ")
         self.tokens = tokens
+        self.id = form_id
         return self
 
     @classmethod
     def from_segmented_str(cls, segmented):
         stripped = segmented.strip(" ")
         self = str.__new__(cls, stripped + " ")
         self.tokens = stripped.split()
+        self.id = None
```

Comment:
I don't think you need this line at all, `__new__` already assigns `form_id`, including when it's None.